functions_framework_tool/templates/helloworld/README.md

```diff
@@ -0,0 +1,111 @@
+# Hello world example
+
+This example handles HTTP GET requests by responding with 'Hello, World!'.
+
+```dart
+// lib/functions.dart
+import 'package:functions_framework/functions_framework.dart';
+import 'package:shelf/shelf.dart';
+
+@CloudFunction()
+Response function(Request request) => Response.ok('Hello, World!');
+```
+
+## Simulate a hosted environment on your own machine
+
+You can run this function example on your own machine using Docker to simulate
+running in a hosted environment.
+
+```shell
+$ docker build -t hello .
+...
+
+$ docker run -it -p 8080:8080 --name app hello
+Listening on :8080
+```
+
+From another terminal:
+
+```shell
+curl http://localhost:8080
+Hello, World!
```

Comment:
You're sending the request an invalid cloudevent. We probably want to error and not just hide the bad request.

Reply:
Sorry, please explain?

Reply:
Oops, this looks good.

I read `CloudEventFunction` rather than `CloudFunction`.

Reply:
Ok, I think we both understand now: @CloudFunction is what we use to mark function intended to be invoked (we inspect the actual shape of the function to know what kind of function it is).